### PR TITLE
Feature/btm 614/contact names config

### PR DESCRIPTION
**Pull request by markschnitziusgovuk**
Changes to read the new vendor service config (see PR with same branch name in di-btm-config) and use it to look up the contract name in all *_curated Views.

### CODE REVIEW

File: src/shared/utils/config-utils/fetch-vendor-service-config.ts [hunk 1]
```diff
@@ -41,4 +42,5 @@ export const isVendorServiceConfigRow = (x: any): x is VendorServiceConfigRow =>
   typeof x.vendor_id === "string" &&
   typeof x.service_name === "string" &&
   typeof x.service_regex === "string" &&
-  typeof x.event_name === "string";
+  typeof x.event_name === "string" &&
+  (x.contract_id === undefined || typeof x.contract_id === "string");
```
[the remark this answers is not in the document]
Comment (markschnitziusgovuk) [reply src/shared/utils/config-utils/fetch-vendor-service-config.ts:46]: No, we actually have several services in prod that we don't currently have contract ids for.  (See the associated PR in di-btm-config.)

[the remark this answers is not in the document]
Comment (markschnitziusgovuk) [reply src/shared/utils/config-utils/fetch-vendor-service-config.ts:46]: I see you've commented there...  We just don't have a contract id for HMPO as far as I know.

File: src/handlers/int-test-support/helpers/payloadHelper.ts [hunk 2]
```diff
@@ -24,6 +25,14 @@ export enum EventName {
   VENDOR_3_EVENT_4 = "VENDOR_3_EVENT_4",
 }
 
+export enum ContractName {
+  vendor_testvendor1 = "C01234",
+  vendor_testvendor2 = "C01235",
+  vendor_testvendor3 = "C01236",
+  vendor_testvendor4 = "FOOBAR1",
+  vendor_testvendor5 = "FOOBAR2",
+}
```
[the remark this answers is not in the document]
Comment (NKannan14) [reply src/handlers/int-test-support/helpers/payloadHelper.ts]: appended with contract_1 with vendor_id

File: integration_tests/tests/contract-table-athena-tests.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import { getContractsFromConfig } from "../../src/handlers/int-test-support/config-utils/get-contract-config-rows";
+
+const configBucket = configStackName();
+
+describe("\n Execute athena query to retrieve contract details details\n", () => {
+  test("retrieved contract details should match with contract csv uploaded in s3 config bucket ", async () => {
+    const queryString = `SELECT * FROM "btm_contracts_standardised"`;
+    const queryResult = await queryAthena(queryString);
+    const queryResultToString = JSON.stringify(queryResult);
+    const queryJsonObj = JSON.parse(queryResultToString);
```
[the remark this answers is not in the document]
Comment (NKannan14) [reply integration_tests/tests/contract-table-athena-tests.ts]: thanks removed stringify and parse as the queryAthena returns string

**Review by MelAtGDS — APPROVED** [review]

**Pull request merged**